### DISCUSSION TITLE
给节点添加上时间段、执行人以及节点进度

### DISCUSSION THREAD

**Discussion by cdRunnet** [hizzgdev/jsmind#624] [General]
你好，现在有一个需求是给每一个节点加上开始时间结束时间以及执行人，我看到了一个自定义渲染节点的，但是没有看到使用示例
附上需求样式图
![企业微信截图_17218792301515](https://github.com/user-attachments/assets/646336a0-8078-4896-abae-a5e1b1ce7a52)

**Comment by cdRunnet** [comment]
我在节点的中点击了html元素但是我不想用户可以编辑这个元素
![image](https://github.com/user-attachments/assets/580ff4a7-0dec-4635-aee1-6d80f85002a8)
![image](https://github.com/user-attachments/assets/8e537207-af0c-4040-9fdc-d7a2aa3fd709)

**Comment by hizzgdev** [comment]
你可以参考[这个文档](https://hizzgdev.github.io/jsmind/docs/zh/options/option.view.custom_node_render.html)了解如何使用 custom_node_render  
如果不想让用户编辑，可以使用[editable选项](https://hizzgdev.github.io/jsmind/docs/zh/options/option.editable.html) 进行设置

另外，如果想要完全实现你截图里的效果，你可能还需要设置[vspace](https://hizzgdev.github.io/jsmind/docs/zh/options/option.layout.vspace.html) 来控制节点的上下间距

**Reply by hizzgdev** [reply]
在使用 custom_node_render 时，你可能需要考虑一下起始时间和执行人的数据从哪里来，可以参考[这个文档里的“节点对象”以及“添加节点”](https://hizzgdev.github.io/jsmind/docs/zh/3.operation.html)，把这些信息放到 `data` 字段里。
在具体的 render 逻辑里，建议使用 `position:relative` 来控制“起始时间和执行人”这个panel 的位置，并限制其高度。

如有疑问请再继续回复。

**Reply by cdRunnet** [reply]
![image](https://github.com/user-attachments/assets/ac53aed7-2130-4356-9b3b-0459ba0fdb36)

custom_node_render: function (jm, element, node) {
console.log(11111111111111)
let emoji = "./123.ico";
element.innerHTML = <img src="${emoji}">${node.topic};
return true; // 表示已经渲染此节点
},
这样写他并没有执行，我并不是不想让用户编辑主题，而是说我不想再用户编辑的时候将html标签进行显示

**Reply by hizzgdev** [reply]
> 这样写他并没有执行

你要不确认一下使用的 jsMind 是什么版本？以及是否js被浏览器缓存了？

**Comment by hizzgdev** [comment]
> 我并不是不想让用户编辑主题，而是说我不想再用户编辑的时候将html标签进行显示

这个话题单独用一个 comment 来讨论吧。

当你提到用户编辑的时候，我想你需要先考虑一下在一开始的时候如何让用户编辑。 jsMind 提供的编辑功能只能修改 topic，如果想在 node 上显示更复杂的数据，建议你自己实现节点编辑的功能，可以想象，时间的输入应该有个 datepicker，执行人的输入应该有一个 userpicker，这些组件还是要以需求为准。但是我想无论如何你一定都有办法把这个编辑 panel 给做出来。

有了这个 panel 后需要做的就是在合适的时候把它显示出来，什么是合适的时候？可能是用户点击了一个 node 的时候，也可能是用户点了某个按钮，也可能是跟页面上的其它元素有交互的时候，总之如果你把 jsMind 当做一个 library 来看待的话，就会有不一样的视角，和很多种实现方法。

**Reply by UmbraCi** [reply]
你好，如果想自定义编辑态从input标签换到textarea，并且支持换行、超链接等自定义的编辑和展示，大概需要怎么做呢？在不修改源码把jsmind当做库的前提下

**Reply by hizzgdev** [reply]
如果要把输入框放在节点上，目前并没有提供API支持这种需求。个人觉得这种复杂的编辑器不应该放在节点上。如果有需求的话就需要修改源码才行。
